File: sdk/assemblyscript/src/assembly/secrets.ts
/*
 * SPDX-FileCopyrightText: © Hypermode Inc. <hello@hypermode.com>
 * SPDX-License-Identifier: Apache-2.0
 */

import * as utils from "./utils";

// @ts-expect-error: decorator
@external("modus_secrets", "getSecretValue")
declare function hostGetSecretValue(name: string): string;

/**
 * Retrieves a secret value from the host environment.
 * Throws if the secret is not found or if an error occurs.
 */
export function getSecretValue(name: string): string {
  const result = hostGetSecretValue(name);
  if (utils.resultIsInvalid(result)) {
    throw new Error("Secret not found");
  }
  return result;
}

File: sdk/go/pkg/secrets/secrets.go
/*
 * SPDX-FileCopyrightText: © Hypermode Inc. <hello@hypermode.com>
 * SPDX-License-Identifier: Apache-2.0
 */

package secrets

import "errors"

// GetSecretValue retrieves a secret value from the host environment.
func GetSecretValue(name string) (string, error) {
	resultPtr := hostGetSecretValue(&name)
	if resultPtr == nil {
		return "", errors.New("secret not found")
	}
	return *resultPtr, nil
}
